File: Back-end/Coffee-Shop/src/main/java/vn/com/ltdt/Coffee_Shop/order/Order.java
package vn.com.ltdt.Coffee_Shop.order;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.UuidGenerator;
import vn.com.ltdt.Coffee_Shop.order.enums.OrderStatus;
import vn.com.ltdt.Coffee_Shop.order.enums.DeliveryType;
import vn.com.ltdt.Coffee_Shop.order.enums.PaymentMethod;
import vn.com.ltdt.Coffee_Shop.user.Customer;
import vn.com.ltdt.Coffee_Shop.user.Employee;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "t_order")
public class Order {
    @Id
    @UuidGenerator
    private String id;
    @NotNull(message = "The payment method cannot be null")
    private PaymentMethod paymentMethod;
    @NotNull(message = "Amount cannot be null")
    private double amount;
    private String note;
    private LocalDateTime orderDate;
    @Enumerated(EnumType.STRING)
    private DeliveryType deliveryType;
    @NotNull(message = "Address cannot be null")
    private String address;
    @OneToMany(mappedBy = "order")
    private List<OrderDetail> details;
    @NotNull(message = "The Customer cannot be null")
    @ManyToOne
    @JoinColumn(name = "customer_id")
    private Customer customer;
    @ManyToOne
    @JoinColumn(name = "admin_id")
    private Employee employee;
    @Enumerated(EnumType.STRING)
    private OrderStatus status;
    private String receivePerson;
    private String receivePhoneNumber;
}

File: Font-end/Mobile/AndroidNative/app/src/main/java/com/ltdt/coffeeshop_android_native/data/domains/Order.kt
package com.ltdt.coffeeshop_android_native.data.domains

import com.ltdt.coffeeshop_android_native.data.remote.OrderDTO
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter


data class Order(
    val id: String? = "NULL CHECK",
    val paymentMethod: PaymentType,
    val amount: Double,
    val note: String = "NULL CHECK",
    var orderDate: LocalDateTime,
    val deliveryType: DeliveryType,
    val address: String = "NULL CHECK",
    val details: List<OrderDetail>,
    val customerId: String = "NULL CHECK",
    val status: OrderStatus,
    val receivePerson: String = "NULL CHECK",
    val receivePhoneNumber: String = "NULL CHECK"
)

fun Order.toOrderDTO(): OrderDTO {
    return OrderDTO(
        id = this.id,
        paymentMethod = this.paymentMethod.name,
        amount = this.amount,
        note = this.note,
        orderDate = this.orderDate.format(DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")),
        deliveryType = this.deliveryType.name,
        address = this.address,
        details = this.details.map { it.toOrderDetailDTO() },
        customerId = this.customerId,
        employeeId = "",
        status = this.status.name,
        receivePerson = this.receivePerson,
        receivePhoneNumber = this.receivePhoneNumber
    )
}
